src/js/services/ng-rankings.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * This is a service that can calculate the rankings.
+ */
+define('services/ng-rankings',[
+    'services/ng-services',
+    'services/ng-stages',
+    'services/ng-teams',
+    'services/ng-score'
+],function(module) {
+    "use strict";
+
+    return module.service('$rankings',
+        ['$stages','$teams', '$score',
+        function($stages, $teams, $score) {
+
+            const EMPTY = '---';
+
+            function group(arr, func) {
+                return arr.reduce((groups, item) => {
+                    let key = func(item);
+                    if(!groups.hasOwnProperty(key)) {
+                        groups[key] = [];
+                    }
+                    groups[key].push(item);
+                    return groups;
+                }, {});
+            }
+
+            function multigroup(arr, funcs) {
+                let currFunc = funcs[0];
+                let result = group(arr, currFunc);
+                if(funcs.length > 1) {
+                    let slicedFuncs = funcs.slice(1);
+                    for(let key in result) {
+                        result[key] = multigroup(result[key], slicedFuncs);
+                    }
+                }
+                return result;
+            }
```

Comment:
This two functions dying belong here: they should be a separate service (maybe even an external component)